pylib/Tools/Fetch/OMPI_Snapshot.py

```diff
@@ -168,16 +169,29 @@ def execute(self, log, keyvals, testDef):
                 # track that we serviced this one
                 self.done.append((tarball_base_name, 1))
                 return
-            # move to that location
-            os.chdir(tarball_base_name)
             # if they want us to leave it as-is, then we are done
             try:
                 if cmds['asis']:
                     status = 0
                     stdout = None
                     stderr = None
+            # If not as-is clear directory and download the tarball
             except KeyError:
-                pass
+                shutil.rmtree(tarball_base_name)
+                # download the tarball - TODO probably need to do a try on these
+                testDef.logger.verbose_print("downloading tarball " + tarball_name + "url: " + download_url)
+                status, stdout, stderr, _ = testDef.execmd.execute(None, ["curl", "-o", tarball_name, download_url], testDef)
+                if 0 != status:
+                    log['status'] = 1
+                    log['stderr'] = "download for tarball " + tarball_name + "url: " + download_url + "FAILED"
+                    return
+                # untar the tarball
+                testDef.logger.verbose_print("untarring tarball " + tarball_name)
+                status, stdout, stderr, _ = testDef.execmd.execute(None, ["tar", "-zxf", tarball_name], testDef)
+                if 0 != status:
+                    log['status'] = 1
+                    log['stderr'] = "untar of tarball " + tarball_name + "FAILED"
+                    return
```

Comment:
Is there a way to remove this code duplication?

Reply:
I tried to pull it out as a independent method but kept getting errors. Finally gave up on it.